009-crd-v1-roadmap.md

```diff
@@ -0,0 +1,108 @@
+For each of these we should make the analogous change that we did for listeners in 0.20.0.
+* Develop the `v1beta2` schema. This must be structural. This must be backward compatible with `v1beta1`, by which I mean any schema-valid `v1beta2` CR must be a schema-valid `v1beta1` CR (though not vice versa).
+* The 0.21.0 CRD will declare `v1alpha1` (served=true, storage=false), `v1beta1` (served=true, storage=false) and `v1beta2` (served=true, storage=true)
+* The operator will access the CRs via the `v1beta2` path (it will be able to observe `v1beta1` resources because the None conversion just changes the apiVersion). 
+* Develop a tool which:
+  * Can rewrite a `v1beta1` resource to an equivalent `v1beta2` resource, allowing users to easily modify their resources (e.g. in git).
+  * Can apply/edit CRs in Kubernetes likewise.
+  * Can update the CRD `spec.versions` to change the served field.
+  * Can update the CRD `status.storedVersions`
+  *  This isn’t completely necessary, but because kubectl doesn’t support the ability to change status it will be very difficult for users to follow the upgrade procedure without it.
+
+#### Users
+
+Users will need to follow a specific procedure to upgrade:
+
+1. Install the 0.21.0 CRDs using `kubectl apply|replace -f` like normal.
+2. Touch each of their CRs, changing the apiVersion to v1beta2 (the tool can do this).
+3. Edit the CRD so that `v1alpha1` and `v1beta1` are both served=false (the tool can do this).
+4. Double check that all their CRs are using v1beta2 and all their apps are still functioning (the tool can do this).
+5. Edit the CRD `status.storedVersions` to remove `v1alpha1` and `v1beta1` (the tool can do this).
+At this point they’re all set for Strimzi 0.22.0
+
+### Strimzi "0.22.0"
+
+This should work on Kubernetes 1.16+ (including 1.21+). This will be the first release using CRD v1 API. 
+
+#### Development
+
+The CRD will be CRD v1 API
+
+#### Users
+
+The upgrade will just be a matter of replacing the CRDs as usual.
+
+### Alternative
+
+Steps 2-5 in the procedure ascribed to Strimzi 0.21, above are technically just preprequisites before installing CRDs which use the v1 API. As such they could equally be done by users as part of the upgrade to Strimzi 0.22. The benefit of doing them sooner is that users would be able to perform those steps during their use of Strimzi 0.21, which might be more convenient.
+
+## Affected/not affected projects
+
+Call out the projects in the Strimzi organisation that are/are not affected by this proposal. 
+
+## Timing
+
+Strimzi 0.21 could be any time before the release of Kubernetes 1.21, but obviously the more time people have to upgrade the better and the more breathing space we have before 1.21 the better to iron out problems people might have in upgrading.
+
+
+
+## Compatibility
+
+Under this proposal Strimzi "0.22" would not support Kubernetes 1.15 or earlier.
```

Comment:
In 0.20.0 you said _This will be the last version of Strimzi which supports Kubernetes 1.11-1.15._ ... so this should be already under 0.21.0 and not 0.2ě.0?